core/src/main/java/com/linecorp/armeria/client/DefaultClientFactory.java

```diff
@@ -77,11 +77,13 @@ final class DefaultClientFactory implements ClientFactory {
 
     static {
         if (DefaultClientFactory.class.getClassLoader() == ClassLoader.getSystemClassLoader()) {
-            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-                if (!shutdownHookDisabled) {
-                    ClientFactory.closeDefault();
+            if (!shutdownHookDisabled) {
```

Comment:
This doesn't do anything because the static initialization block is always called before static method(e.g. `disableShutdownHook0()` here).
Let's just catch the exception instead.

Reply:
Thanks for explain, updated 🙇

Reply:
> This doesn't do anything because the static initialization block is always called before static method(e.g. disableShutdownHook0() here).

Question) Would users still want to disable automatic shutdown to a later timing of their choosing by calling `ClientFactory#disableShutdownHook`? or is https://github.com/line/armeria/pull/1472 not an issue anymore?